.github/workflows/pkg.yml

```diff
@@ -33,15 +42,86 @@ jobs:
 
           # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
           # skip-build-cache: true
+
+      - name: Print lint report artifact
+        if: always()
+        run: test -f golangci-lint-report.xml && cat golangci-lint-report.xml || true
+
+      - name: Upload lint report artifact
+        if: always()
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
+        with:
+          name: golangci-lint-report
+          path: golangci-lint-report.xml
+
   build-test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
         with:
           go-version-file: 'go.mod'
-      - name: build
+
+      - name: Build
         run: go build -v ./...
-      - name: test
+
+      - name: Unit Tests
+        run: go test -v ./... -coverpkg=./... -coverprofile=coverage.txt $1 | tee $OUTPUT_FILE
+      
+      - name: Upload Go test results
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: go-test-results
+          path: |
+            ./output.txt
+            ./coverage.txt
+  
+  race-tests:
```

Comment:
Why is this separate? We don't get any go build caching this way.

Reply:
Moved the race flag under unit tests.